packages/react-instantsearch-hooks-dom/src/widgets/__tests__/RefinementList.test.tsx

```diff
@@ -0,0 +1,3462 @@
+                  type="checkbox"
+                  value="Incipio"
+                />
+                <span
+                  class="ais-RefinementList-labelText"
+                >
+                  Incipio
+                </span>
+                <span
+                  class="ais-RefinementList-count"
+                >
+                  320
+                </span>
+              </label>
+            </li>
+            <li
+              class="ais-RefinementList-item"
+            >
+              <label
+                class="ais-RefinementList-label"
+              >
+                <input
+                  class="ais-RefinementList-checkbox"
+                  type="checkbox"
+                  value="KitchenAid"
+                />
+                <span
+                  class="ais-RefinementList-labelText"
+                >
+                  KitchenAid
+                </span>
+                <span
+                  class="ais-RefinementList-count"
+                >
+                  318
+                </span>
+              </label>
+            </li>
+            <li
+              class="ais-RefinementList-item"
+            >
+              <label
+                class="ais-RefinementList-label"
+              >
+                <input
+                  class="ais-RefinementList-checkbox"
+                  type="checkbox"
+                  value="Whirlpool"
+                />
+                <span
+                  class="ais-RefinementList-labelText"
+                >
+                  Whirlpool
+                </span>
+                <span
+                  class="ais-RefinementList-count"
+                >
+                  298
+                </span>
+              </label>
+            </li>
+          </ul>
+        </div>
+      </div>
+    `);
+
+    const firstCheckbox = container.querySelector(
+      '.ais-RefinementList-checkbox'
+    ) as HTMLInputElement;
+
+    expect(firstCheckbox).not.toBeChecked();
+
+    userEvent.click(firstCheckbox);
+
+    await wait(0);
+
+    expect(firstCheckbox).toBeChecked();
+    // Once on load, once on check.
+    expect(client.search).toHaveBeenCalledTimes(2);
+    expect(client.search).toHaveBeenLastCalledWith(
+      expect.arrayContaining([
+        expect.objectContaining({
+          params: expect.objectContaining({
+            facetFilters: [['brand:Insignia™']],
+          }),
+        }),
+      ])
+    );
+  });
+
+  test('enables conjunctive faceting', async () => {
+    const client = createSearchClient({ search });
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={client}>
+        <RefinementList attribute="brand" operator="and" />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await wait(0);
+
+    const [checkbox1, checkbox2] = [
+      ...(container.querySelectorAll(
+        '.ais-RefinementList-checkbox'
+      ) as NodeListOf<HTMLInputElement>),
+    ].slice(0, 2);
```

Comment:
Could we use `getByRole()` with `name` for this?

Reply:
We could, but this way we also ensure the elements with the right class names are present and they're the ones triggering a change.